kymatio/scattering2d/filter_bank.py

```diff
@@ -36,35 +36,39 @@ def filter_bank(M, N, J, L=8):
     filters = {}
     filters['psi'] = []
 
-    offset_unpad = 0
     for j in range(J):
         for theta in range(L):
             psi = {}
             psi['j'] = j
             psi['theta'] = theta
             psi_signal = morlet_2d(M, N, 0.8 * 2**j,
                 (int(L-L/2-1)-theta) * np.pi / L,
-                3.0 / 4.0 * np.pi /2**j, 4.0/L, offset=offset_unpad)
+                3.0 / 4.0 * np.pi /2**j, 4.0/L)
             psi_signal_fourier = fft2(psi_signal)
+            # drop the imaginary part, it is zero anyway
+            psi_signal_fourier = np.real(psi_signal_fourier)
             for res in range(min(j + 1, J - 1)):
                 psi_signal_fourier_res = periodize_filter_fft(
                     psi_signal_fourier, res)
-                psi[res] = torch.FloatTensor(
-                    np.stack((np.real(psi_signal_fourier_res),
-                    np.imag(psi_signal_fourier_res)), axis=2))
+                # add a trailing singleton dimension to mark it as non-complex
+                psi_signal_fourier_res = psi_signal_fourier_res[..., np.newaxis]
+                psi[res] = torch.FloatTensor(psi_signal_fourier_res)
                 # Normalization to avoid doing it with the FFT.
                 psi[res].div_(M*N// 2**(2*j))
             filters['psi'].append(psi)
 
     filters['phi'] = {}
-    phi_signal = gabor_2d(M, N, 0.8 * 2**(J-1), 0, 0, offset=offset_unpad)
+    phi_signal = gabor_2d(M, N, 0.8 * 2**(J-1), 0, 0)
     phi_signal_fourier = fft2(phi_signal)
+    # drop the imaginary part, it is zero anyway
+    phi_signal_fourier = np.real(phi_signal_fourier)
     filters['phi']['j'] = J
     for res in range(J):
         phi_signal_fourier_res = periodize_filter_fft(phi_signal_fourier, res)
-        filters['phi'][res] = torch.FloatTensor(np.stack(
```

Comment:
I didn't remember there was some tensor used here; it'd be better to employ some numpy structures for the reason mentionned above